src/Messages/TSMessagesManager.m

```diff
@@ -40,6 +40,36 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+static TSMessagesManager *shareMessagesManager = nil;
+
+id shareMessagesManagerLock() {
+    static id shareMessagesManagerLock = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        shareMessagesManagerLock = [NSObject new];
+    });
+    return shareMessagesManagerLock;
+}
+
+TSMessagesManager *GetShareMessagesManager() {
+    @synchronized (shareMessagesManagerLock()) {
+        OWSCAssert(shareMessagesManager);
+        
+        return shareMessagesManager;
+    }
+}
+
+void SetShareMessagesManager(TSMessagesManager *value) {
+    @synchronized (shareMessagesManagerLock()) {
+        OWSCAssert(!shareMessagesManager);
+        OWSCAssert(value);
+        
+        shareMessagesManager = value;
+    }
+}
```

Comment:
Not the most elegant solution, but it'll allow us to inject TSMessagesManager's dependencies in `PushManager`.